Bottom: 8241b93e1ff327d3f5126be15054f102edf56889
Top:    be69ee106f6885ceef786f3070dcc0a9796cc1dc
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-12-21 15:07:48 +0800

Refresh of tuntap-rename-struct

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index c6bbed5..6ab9e4e 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -238,7 +238,6 @@ struct tun_struct {
 	struct tun_pcpu_stats __percpu *pcpu_stats;
 	struct bpf_prog __rcu *xdp_prog;
 	struct tun_prog __rcu *steering_prog;
-	struct tun_prog __rcu *filter_prog;
 };
 
 static int tun_napi_receive(struct napi_struct *napi, int budget)
@@ -2071,7 +2070,6 @@ static void tun_free_netdev(struct net_device *dev)
 	tun_flow_uninit(tun);
 	security_tun_dev_free_security(tun->security);
 	__tun_set_ebpf(tun, &tun->steering_prog, NULL);
-	__tun_set_ebpf(tun, &tun->filter_prog, NULL);
 }
 
 static void tun_setup(struct net_device *dev)
@@ -2853,10 +2851,6 @@ static long __tun_chr_ioctl(struct file *file, unsigned int cmd,
 		ret = tun_set_ebpf(tun, &tun->steering_prog, argp);
 		break;
 
-	case TUNSETFILTEREBPF:
-		ret = tun_set_ebpf(tun, &tun->filter_prog, argp);
-		break;
-
 	default:
 		ret = -EINVAL;
 		break;
